apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.html.ts

```diff
@@ -0,0 +1,92 @@
+import { z } from 'zod';
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { isEnterprise } from '../../lib/license';
+
+export const InstitutionSchema = z.object({
+  id: z.string(),
+  short_name: z.string(),
+  long_name: z.string(),
+  display_timezone: z.string(),
+  uid_regexp: z.string().nullable(),
+  authn_providers: z.array(z.string()),
+});
+type Institution = z.infer<typeof InstitutionSchema>;
+
+export function AdministratorInstitutions({
+  institutions,
+  resLocals,
+}: {
+  institutions: Institution[];
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head'); %>", {
+          ...resLocals,
+          pageTitle: 'Institutions',
+        })}
+      </head>
+      <body>
+        <script>
+          $(function () {
+            $('[data-toggle="popover"]').popover({ sanitize: false });
+          });
+        </script>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'admin',
+          navSubPage: 'institutions',
+        })}
+        <main id="content" class="container-fluid">
+          <div id="institutions" class="card mb-4">
+            <div class="card-header bg-primary text-white d-flex align-items-center">
+              Institutions
+            </div>
+            <div class="table-responsive">
+              <table class="table table-sm table-hover table-striped">
+                <thead>
+                  <tr>
+                    <th>Short name</th>
+                    <th>Long name</th>
+                    <th>Timezone</th>
+                    <th>UID regexp</th>
+                    <th>Authn providers</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  ${institutions.map(
+                    (inst) => html`
```

Comment:
To match the suggested change to the schema:
```suggestion
                    ({institution, authn_providers}) => html`
```
The naming convention I've been using here is that whenever I have a variable called `table_name` (singular) then I know it's a row from the table, and `table_names` (plural) means it's an array of rows from the named table. I avoid using abbreviations unless really necessary, to cut down on the cognitive overhead. If I have something else (like an array of authn provider names) then I call it something else (e.g., `authn_provider_names`) but I try and always use the full types wherever possible -- I've found it makes it much easier/faster to modify code.

Reply:
This is courtesy of a mindless copy/paste from EJS, I've definitely been avoiding abbreviations otherwise. I'll fix this.

Reply:
Done in d935d41.